### PR TITLE
perf(column): avoid counting when max signs are removed from a range

### PR DESCRIPTION
**Pull request by luukvbaal**

### CODE REVIEW

**Review by luukvbaal — COMMENTED** [review]
Also fixes two issues and modified some comments/conditions.

File: src/nvim/decoration.c [hunk 1]
```diff
@@ -847,8 +851,8 @@ static void buf_signcols_validate_range(buf_T *buf, int row1, int row2, int add)
       count++;
       if (mt_paired(mark)) {
         MTPos end = marktree_get_altpos(buf->b_marktree, mark, NULL);
-        for (int i = mark.pos.row; i < MIN(row2, end.row); i++) {
-          overlap[row2 - i]++;
+        for (int i = mark.pos.row + 1 - row1; i < MIN(row2, end.row); i++) {
```
Comment (luukvbaal) [inline src/nvim/decoration.c:854]: This loop range made no sense actually, was just the first thing I tried and didn't happen to fail any tests. Used my 🧠 to verify correctness and added a test to `decorations_spec.lua` that failed.

**Comment by justinmk** [comment]
Coverity warning:

```
*** CID 470331:  Error handling issues  (CHECKED_RETURN)
/src/nvim/decoration.c: 824 in buf_signcols_validate_range()
818     
819       // Allocate an array of integers holding the overlapping signs in the range.
820       assert(row2 >= row1);
821       int *overlap = xcalloc(sizeof(int), (size_t)(row2 + 1 - row1));
822     
823       // First find the number of overlapping signs at "row1".
>>>     CID 470331:  Error handling issues  (CHECKED_RETURN)
>>>     Calling "marktree_itr_get_overlap" without checking return value (as is done elsewhere 4 out of 5 times).
824       marktree_itr_get_overlap(buf->b_marktree, currow, 0, itr);
825       while (marktree_itr_step_overlap(buf->b_marktree, itr, &pair)) {
826         if (!mt_invalid(pair.start) && pair.start.flags & MT_FLAG_DECOR_SIGNTEXT) {
827           overlap[0]++;
828         }
829       }

```

**Comment by luukvbaal** [comment]
False negative but we can suppress it I guess.

**Comment by justinmk** [comment]
Yeah, worth adding a `(void)` since doing it manually in coverity can get stale if the code moves around.

**Pull request merged**